### ISSUE TITLE
boost with comment

### ISSUE THREAD

**Issue by cgubi** [mastodon/mastodon#14597]
The option "boost with comment" could useful to add insight, context and remarks to boosted content

### Pitch

This feature should be similar to the "retweet with comment" feature of Twitter. If you click on the "retweet" icon you get a dropdown menu allowing you to choose between "retweet" and "retweet" with comments.

To make it more straightfoward we could boost with click and boost with comment with right click, and on mobile device we could boost with tap and boost with comment with long press.

### Motivation

I think that this feature could be useful for the entire mastodon community, to facilitate migration of twitter user, and will introduce an added value to the UX, allowing to comment and highlight contents, also with a warning/reporting attitude for controversial contents, that could be boosted with a comment like "guys, this sound fake to me, what do you think?". Unfortunately, if we have the boost as only option this kind of sharing is not allowed and boosting will have always a positive, supporting meaning, where we could give space to more nuanced comments when boosting.

**Comment by ThibG** [comment]
This is a duplicate of #309. I'm going to let that open for now, but I don't think @Gargron has changed his mind (https://github.com/tootsuite/mastodon/issues/309#issuecomment-469260847)

**Comment by cgubi** [comment]
> This is a duplicate of #309. I'm going to let that open for now, but I don't think @Gargron has changed his mind ([#309 (comment)](https://github.com/tootsuite/mastodon/issues/309#issuecomment-469260847))

Thanks @ThibG, i read all the rational behind the closure of the issue, but after 4 years the audience of social media changed and users habits as well. The main usage for comments to retweets is not trolling or summon people against a specific user or content, but we can find this feature in the toolbox of debunkers that tweet controversial quotes with their comments and links to debunking sources, of human rights defenders that highlight and comment questionable quotes of oppressive governments, comments to retweets and also used for debating pseudoscientific claims, so maybe it may worth reconsidering the issue also for toots and boosts.

Thanks for your attention.

**Comment by bluss** [comment]
Is there an alternative here? For example, if instead you'd make a reply to the original post, and somehow could make that reply visible in your regular timeline, and to those that don't follow the post you reply to. Like an "open reply"?

**Comment by ThibG** [comment]
> Is there an alternative here? For example, if instead you'd make a reply to the original post, and somehow could make that reply visible in your regular timeline, and to those that don't follow the post you reply to. Like an "open reply"?

You can already boost your reply, to get that behavior.

**Comment by jdaviescoates** [comment]
I just wanted to quote boost something and realised I couldn't so came here to see people have been asking it for years in various different issue.

Funnily enough Twitter are rolling out a UX change so that even if you just want to RT the user interface pops up the Quote RT box (I guess to encourage more Quote RTs).

**Comment by realpixelcode** [comment]
I'd definitely like to have this feature too – not only because it's a nice option to have but also because former Twitter users moving to Mastodon should feel comfortable so that Mastodon may gain new users.

**Comment by KuJoe** [comment]
I'd like to share a short thread where another Mastodon user and I discuss a possible solution to this without just being a "Quote Retweet" clone that Mastodon is trying to avoid:

https://mindly.social/web/@KuJoe/108336435820590456

**Comment by saschanaz** [comment]
> I'd like to share a short thread where another Mastodon user and I discuss a possible solution to this without just being a "Quote Retweet" clone that Mastodon is trying to avoid:
> 
> https://mindly.social/web/@KuJoe/108336435820590456

That sounds like just a visual tweak, still with a list of quotes visible to everyone and still notifies every time a quote happens. And I think the the list actively encourages dunking on the quoted post by showing other dunks, and the notification exposes the poster to all the attacks.

That said, I do want quotes but with those two "features" excluded.

**Comment by KuJoe** [comment]
> That sounds like just a visual tweak, still with a list of quotes visible to everyone and still notifies every time a quote happens. And I think the the list actively encourages dunking on the quoted post by showing other dunks, and the notification exposes the poster to all the attacks.

My apologies. Having not seen any of these dunks/attacks before I'm probably approaching this from a different direction. I'm still trying to wrap my head around how quotes are being used negatively which is probably why my thought process is unable to account for that abusive nature.

**Comment by Truhe2342** [comment]
Quote: Talk about other people
Reply: Talk with other people
Boost: Share content of other people

Quote is very often used to talk bad about other peoples content or to write an angry rant on other peoples content. It generates a negative feeling in your timeline and the timeline of those, who follow you, as proven by Twitter.

You can still just post and link the original post. Yes, that's more inconvenient – by choice – to prohibit the negative usage of quotes.

**Comment by saschanaz** [comment]
> You can still just post and link the original post. Yes, that's more inconvenient – by choice – to prohibit the negative usage of quotes.

Except you don't know how the preview would look before posting it (#18224) and people won't immediately be able to read the preview either (#4486, #12738).

**Comment by SkaveRat** [comment]
I feel not adding this feature by choice is throwing out the baby with the bathwater.

Any "negative-quoting" can also be done now. "Yes, that's more inconvenient", but still very much possible.

Just now we also don't have any of the positive things a quote-post with preview gets us

**Comment by eloquence** [comment]
Has it been considered to offer the equivalent of quote tweets _only in replies_, a kind of "cite a post" feature? 

Here's why I think this might not be a completely dumb idea:

- Prevents populating the main timeline with "dunking" type posts, the original motivation against this functionality

- Allows users to cite relevant posts in discussion in a user-friendly manner that could be linked to other functionality such as notifications (your post X was cited in reply Y).
 
- For folks who really feel a need to call out or document bad behavior for legitimate reasons, it does not completely foreclose that possibility.

Of course, it would leave open some potential for misuse, and replies with embedded posts might have to be made "unboostable" to avoid circumventing the replies-only restriction. With that and other trade-offs in mind, does it sound to folks who desire quote-boosts that it might offer a reasonable middle ground?

**Comment by lolgesten** [comment]
Well thought out motivations why boosting with comments is a good idea by Rich Felker (@dalias@hachyderm.io) here: https://dulc.es/@dalias@hachyderm.io/109355822002773101

**Comment by thefuturebird** [comment]
We've been having a lot of conversations about how quote boosts could be used. I have been using the hashtag "#MissedQuoteBoost" to keep a catalog of times when a quote boost would have been useful.  I started doing this because I kept encountering people who were convinced that quote boost could *only* be used for "dunking" -- that it was a bad feature that creates bad interactions and that simply has not been my experience in over a decade using twitter. 

Others have joined in doing this. From translation, to adding additional context to ongoing live events the uses are many and varied. There are so many objectively positive uses for quote boost:

https://sauropods.win/tags/missedquoteboost

People keep repeating this "wisdom" that "_quote tweets enable toxic behavior_" but how exactly do we know that to be true? I have been attacked with racial slurs on mastodon just like on twitter. I don't see how it changes anything. Especially when bad actors can use screenshots, or their own uniquely configured servers to effectively use them anyways. It really seems like this is about shutting down more ordinary users, users who aren't bad actors.  It seems like something else is going on.

It seems like it's about keeping things quiet and polite. **Avoiding conflict isn't the same thing as preventing abuse.** In fact, minimizing conflict and keeping things polite can work in opposition to prevention of abuse.  Everyone knows that quote tweets were an effective tool to call out racism on twitter. A way to avoid isolation. A way to show what happened directly so that people would believe you. This use of a quote isn't peaceful or calm, and it's much much more rare than the sharing and collaboration that makes up most quote tweets-- but it seems like stopping this from happening merits a policy that also prevents everything else. 

And speaking of conflict, for better or worse this issue has come to represent something more to many users. When a social network that has existed for a long time experiences sudden growth there will be some inevitable friction between the newer and older users. Should mastodon accommodate the needs of the new users, or do new users need to change to accommodate the network? Obviously, this isn't a binary. Both parties should adapt. But, much of how mastodon is perceived, and much of the subtext around these conversations has taken on a rather nasty tone. I've been told:

_" The twitter evacuees want it too much, and sugar coat their reasons, but its what they want, and people will start implementing this stupid mal-feature."_

_"Stop trying to ruin this place."_

_"the people asking for it [quote-boost] the loudest were the ones abusing it a lot on Twitter.  They are demanding their bad habit be accommodated."_

And other similar comments that cast users who have come from twitter in a bad light, presume we are being dishonest and allude to us as some kind of chaotic force that needs to be controlled in this new environment. It's a really bad look, for ya'll. Especially when the proposed implementation is one where everyone has the option to _opt in_.  That is, no one will have their posts quoted using this feature who doesn't want to have them quoted. This, then, becomes about policing the way that _other people_ interact with each other.  I kind of "no I know what's best for you" It is pretty condescending. 

You see, I don't really think mastodon needs quote-boost to thrive. It would be better, helpful, positive, but users are creative and will work around it. We will find new ways to communicate. What I'm concerned about **now** is what this restriction _has come to represent._ 

Whose concerns matter?
Who is welcomed?
Who is supported?
Who isn't?

I would say, ignore these things at your own peril if you care about continuing to grow into a diverse and thriving network. If this issue is simply ignored. If the devs stay within what seemed obvious "obviously quote tweets are bad. obviously they are part of that fractious and overly lively world of twitter. obviously if people from twitter want them it's because they're trying to ruin this place etc." it will become one of those things that's cited when people write articles about "How Mastodon had the Chance to be Better than Twitter: _and why it didn't happen_."

It won't be the only reason or even the biggest one, but it will be _an example._ 

So, it's not even about if quote-boost is implemented or not. (Maybe it's too difficult to implement. Maybe there are more pressing issues.) It's about the assumptions implied by these discussions. It's about what it means when someone hears "well the devs decided not to implement that because it leads to toxic interactions" and we are talking about a feature that is synonymous with so-called "call out culture."  

One could equally say that allowing replies to posts allows toxic interactions. (All of the slurs I've been called were in replies.) No one would talk about banning replies to prevent toxic interactions. We recognize that replies can be used in many ways.  But quote boost? the "call out culture" feature? That must be avoided. Because Mastodon will be a more polite and calm sort of place, a conflict-adverse sort of place. There isn't any need to study the feature objectively and see how it was really used-- even if it turns out it wasn't even really used for "call outs" very often -- the mere association with those kinds of interactions makes it something Mastodon wants no part of. And by extension Mastodon wants no part of the communities associated with this feature. This subtext only grows louder and louder the more I investigate this topic. 

I encourage you to go to twitter and look at how people used the feature. Take a look at my posts if you like:

https://twitter.com/futurebird

Look at how people really used quote tweets. Consider how people want to use quote boost here. Seriously consider where the assumptions people make about this feature are coming from and what it says about what Mastodon prioritizes: minimizing conflict? Or minimizing harm?

**Comment by d3cline** [comment]
If it can be used as a politically motivated weapon its not a feature we should want. Even if your political viewpoints think its a good weapon. The last thing we need is more antagonism or any antagonism amplifying features. If the feature is to be added it needs to be added in such a way as to disarm dog-piling attacks. And to disarm political disagreements. I have been the victim of this attack and been called a fascist and xenophobe and have been put on #fediblock for it. And it was because I have centrist worldviews and see all humanity as one species. It happened again today when engaging with thefuturebird on mastodon itself. The following servers openly and wantonly perform these kinds of attacks, 
rage.love 
kolektiva.social 
And have been blocked from my instance. Although instance blocking is effective at this level, we should be very cautious as to add a core feature which makes it a strategy everyone sees as accepted. 
The event of being put on feddiblock caused my bipolar disorder to trigger a depressed state. Suicidal ideations. I am lucky to have the treatment I need to deal with it. But some people don't. The role of admin is already difficult enough to keep peace without more antagonism. Politically motivated and fueled antagonism. Its no joke or laughing matter. 
If the feature is only going to take into account the political needs of one group and not everyone its suspect to me. If it is used to dog pile and trigger people with mental disorder into harm its a big NO. Twitter still exists, we don't need to mirror all their features and problems. 

TLDR it might have done X, Y, or Z on a centralized system, but what will it do on a decentralized system? Could it cause so much division as to harm the federated network? Do we want more echo chambers?  Can it be implemented in a way which mitigates abuse?

**Comment by pcantrell** [comment]
I too would like a “boost with comment” feature, for many of the same reasons @thefuturebird lays out above. Context can so often extend the audience of the post in a _positive_ way: “here’s how this is relevant;” “pay attention to this part;” “here’s how this other thing is connected to the conversation at hand.” I want a feature that supports that.

Beyond the utility of the feature itself, it’s important to pay attention to the larger argument in the second half of the post above: thoughtfully incorporating feedback from new users (or not) is not just software development strategy; it is a statement of community values. It is a test of whether the platform can evolve and survive. And I do want Mastodon to evolve and survive.

For that to happen, the Mastodon developer community needs to be able to incorporate user feedback as the user community expands. That doesn’t mean implementing all user requests at face value. It does mean **listening** to what people say, **valuing** the feedback and the people providing it, **rethinking** past assumptions, and **incorporating** the new perspective thoughtfully.

“We don’t do that and you’re wrong, go away” is the sound of a doomed platform. That’s not what I want for Mastodon. There’s a good design solution here, I’m sure of it.

---

All that said, I’m sensitive to concerns about how such a feature might encourage dogpiling and abuse.

Folks brush off the idea of using UX tweaks to solve the problem of abusive use, but I’m optimistic about that approach. Imagine a feature called “Boost with comment” that uses that terminology on both the posting _and_ reading sides: Imagine that a user named **BlueWave2024** wants to dunk on Mitch McConnell. And imagine that they know that if they boost his post with a comment, then what their followers will see is:

 **BlueWave2024 boosted Mitch McConnell**

That alone, just that much, seems to me like it would be a powerful disincentive to use the feature for hostility.

**Comment by danmarce** [comment]
To be honest, it is a useful function. If you reply to a post, your followers might miss the original post thinking "what is he talking about". If you boost and then post something commenting, the ideas might become unclear, separated if your follower has a busy timeline, what is he talking about" again. 

Quote boost solves this, shows with ease what are you talking about, and keeps the conversation there.

I mean, just as youtube videos are embedded in the post a link to a post can show the actual post (in theory this could work also with tweets, to bring the conversation to mastodon, but that is a separated issue). 

**People does this now taking a screenshot and commenting**, the users need the functionality and use the worst alternative possible: because doing that, keeps the original poster in the dark about somebody "boosting" his post (so, you gain hate, or love and never know about it, now even having a chance to comment back)

**Comment by bengartner** [comment]
@thefuturebird thank you so much for your post, it's very helpful context for me to understand and has me thinking of new ways the feature could work.

My particular software expertise does not put me in a position to implement anything on Mastodon, but in my experience implementing software features, it's extremely important for the devs to thoroughly understand the problem statement fully before committing to a particular solution. I would highly recommend starting a discussion of the need for users to call out racism in a new thread so that finding a solution is the explicit goal of the thread and the QT can be discussed in the context of alternatives.

I also do not have any context or authority to speak to the subtext of the discussion, or whether this community will be uninterested in solving the problem you've presented. I don't see it as a consideration in this thread from 2016: https://github.com/mastodon/mastodon/issues/309

> One could equally say that allowing replies to posts allows toxic interactions. (All of the slurs I've been called were in replies.) No one would talk about banning replies to prevent toxic interactions. We recognize that replies can be used in many ways. But quote boost? the "call out culture" feature? That must be avoided. Because Mastodon will be a more polite and calm sort of place, a conflict-adverse sort of place. There isn't any need to study the feature objectively and see how it was really used-- even if it turns out it wasn't even really used for "call outs" very often -- the mere association with those kinds of interactions makes it something Mastodon wants no part of. And by extension Mastodon wants no part of the communities associated with this feature. This subtext only grows louder and louder the more I investigate this topic.

The difference in Mastodon's moderation model and who pays for server costs is driving some of the technical discussions here. Different parts of Mastodon moderate differently. My understanding is that moderators are more responsive to banning entire parts of Mastodon that allow overt racism or bad faith actors.

However, the need to call out and discuss is a different need in many subtle ways, and may require different levels of moderation support. Imagine a "call out" button that is more explicit to the use case your community values. Would it imply a request for moderation? Or would it actually imply the opposite? A request that the moderator leave the content in place so it can be discussed? Would it imply a content warning? Does the expected behavior differ if the post is on a different server whose moderation policy is different? Is the feature intended to be part of a feedback loop with moderators? Or does calling out not imply a corrective action and the need is to commiserate with your community?

Is there a need to call out and discuss an entirely different server, if the moderation there is what needs to be called out and not one individual toot?

One of the technical limitations of Mastodon's server model is that different phone apps or individual servers ultimately control how the data is displayed. So any QT feature will look different to different users, the same way Outlook shows reply quotes different than gmail chooses to. Clarifying why the visual display is important will help developers clarify what the technical constraints are. 

One of my frustrations as a software developer is seeing sites like twitter and facebook but up unnecessary barriers and suck all the creative oxygen out of the room. We really have no idea what the optimal solutions are once we take engagement incentives and advertising revenue outside of the equation. Ultimately, the goal is to find a way to effectively call out racism. The solution may have identical functionality as QTs, but with different cultural context. Or it may involve more input from moderators and look entirely different. Different branches of Mastodon will almost certainly try different things and we'll have some debate about which succeed. In all cases, the discussion of the problem is much more important than the solution. 

There are different assumptions at play about why QTs are valuable, and discussions in the solution space require assumptions about what problems we're trying to solve and what problems may be created. This is a common pattern in technical communication. By clarifying the problem statement without the context of the presumed solution, we are more likely to identify the conflicting assumptions and gain some empathy/understanding.

Admittedly, we may not get to a solution faster that way. But in my experience, implementing a relatively straightforward solution like QTs, without understanding the context of the more complicated problem it tries to solve, almost always leads to unintended consequences.

**Comment by growf** [comment]
Can we just adjust existing functionality: automatically add a preview of the post being replied to if we boost our own reply to something?

This prevents quoting without engaging with the original post and also prevents quoting a post with replies disabled.

**Comment by realpixelcode** [comment]
Proposal:

- Let everyone decide themselves by whom they can be quoted, e.g. followers, following, same-server, anyone. Only quoting, if the author allows it.
- Ban “screenshot boosts” in order to prevent people from talking behind someone's back by circumventing the quote boost notification.

Isn't that an option?

**Comment by h-2** [comment]
I just got aware of this issue! Note that we are having a long discussion over here, as well:
https://github.com/mastodon/mastodon/issues/20673

[advertising-mode]I think it is the open issue with the most upvotes in regard to "quote-boosting", so please give it a :+1: , if you would like to see this feature[/advertising-mode]

**Comment by davemacdo** [comment]
I'll add a couple of things to this discussion that I think are important consideration that I don't think I've seen come up (at least not in this thread): 

1. One of the reasons QTs are/were so easy to use abusively on Twitter is the algorithmic timeline. By rewarding """engagement""", it created a feedback loop of outrage. Since we _don't_ have any kind of algorithmic ranking for timelines, think quote-post abuse would automatically be much less common. 
2. The screenshot workaround that is currently being used has lots of problems. As has been pointed out above, the original post author is not notified. An issue of equal (and I might even argue greater) importance is that without some extra effort on the part of the user, the screenshot is not accessible to screen-reader users. 
3. QTs on Twitter originated as a user behavior, much like @ replies, RTs, and hashtags, that were then adopted as richer features of the platform. Regardless of what the Mastodon contributors think about quote-posts, users will do them. The options are not have quote-posts or not have quote-posts. The options are have quote-posts in a way that is harmful and uncontrolled, or have quote-posts implemented in a way that gives users (both quoter and quotee) control over how they are represented in a conversation, and a way that allows moderation and discourages abuse.

**Comment by brendanjones** [comment]
Putting on my product owner hat, I thought it a good idea to separate out quote posting features and any settings to control quote posting. Now this issue can narrow scope to focus on quote post implementation. https://github.com/mastodon/mastodon/issues/25735

**Comment by diondiondion** [comment]
Consent-respecting quote posts were introduced in [Mastodon 4.5](https://github.com/mastodon/mastodon/releases/tag/v4.5.0) so we're closing this issue now. If you have any concerns or feedback about the new quote posts, please open a new issue.

You can learn more about Quote posts in our [blog post](https://blog.joinmastodon.org/2025/09/introducing-quote-posts/), our [FAQ](https://docs.joinmastodon.org/user/quote-posts/#faq), and our [developer implementation guide](https://docs.joinmastodon.org/client/quotes/).
